src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/get_options_list_control_factory.tsx

```diff
@@ -151,8 +138,23 @@ export const getOptionsListControlFactory = (): DataControlFactory<
           selectionsManager.api.setSelectedOptions(undefined);
           selectionsManager.api.setExistsSelected(false);
           selectionsManager.api.setExclude(false);
+          selectionsManager.api.setSort(OPTIONS_LIST_DEFAULT_SORT);
           temporaryStateManager.api.setRequestSize(MIN_OPTIONS_LIST_REQUEST_SIZE);
-          sort$.next(OPTIONS_LIST_DEFAULT_SORT);
+        });
+
+      // we do not want to delay the embeddable creation waiting for this, so do not await promise
+      const allowExpensiveQueries$ = new BehaviorSubject<boolean>(true);
+      coreServices.http
+        .get<{
+          allowExpensiveQueries: boolean;
+        }>('/internal/controls/getExpensiveQueriesSetting', {
+          version: '1',
+        })
+        .catch(() => {
+          return { allowExpensiveQueries: true }; // default to true on error
+        })
+        .then(({ allowExpensiveQueries }) => {
+          if (!allowExpensiveQueries) allowExpensiveQueries$.next(false);
```

Comment:
As noted in https://github.com/elastic/kibana/pull/245588#discussion_r2620949118, we will be removing `allowExpensiveQueries` - so while moving the logic from the control group (hence only requiring a single request) to the individual control (hence requiring a request **per control**) will result in a slight performance degradation, this is only temporary.